### PR TITLE
Add iota snap

### PR DESCRIPTION
**Pull request by mariogrip**
This requires this patch of snapcraft to work: https://github.com/ubuntu-core/snapcraft/pull/579

### CODE REVIEW

**Comment by dholbach** [comment]
*[iota/snapcraft.yaml, line 12 \[r1\]](https://reviewable.io:443/reviews/ubuntu/snappy-playpen/76#-KKmXNDoKnE-Efc-2HnC:-KKmXNDpxQMVKisNTChV:-1602625031) ([raw file](https://github.com/ubuntu/snappy-playpen/blob/d67f944f0c801ee04a516eb4575c232d25d8f708/iota/snapcraft.yaml#L12)):*
> ```YAML
> parts:
>   iota:
>     plugin: rust
> ```

We're still blocked here.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/ubuntu/snappy-playpen/76)*

**Comment by tachyons** [comment]
Any updates ? rust plugin is already merged to snapcraft

**Comment by dholbach** [comment]
Thanks for the ping, I restarted the CI run.

File: iota/snapcraft.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+description: Iota is a terminal-based text-editor written in Rust.
+
+apps:
+    iota:
+        command: iota
+
+parts:
+  iota:
+    plugin: rust
+    source-type: git
```
Comment (dholbach) [inline iota/snapcraft.yaml]: This line shouldn't be necessary as the URL below ends with `.git`.

File: iota/snapcraft.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+name: iota
+version: 0.1
+summary: Iota is a terminal-based text-editor written in Rust.
+description: Iota is a terminal-based text-editor written in Rust.
+
```
Comment (dholbach) [inline iota/snapcraft.yaml]: It'd be good to define `confinement` and `grade` in here.

File: iota/snapcraft.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+apps:
+    iota:
+        command: iota
```
Comment (dholbach) [inline iota/snapcraft.yaml]: Are no `plugs` required? `[home]` maybe?

**Review by dholbach — CHANGES_REQUESTED** [review]
Some small bits need to be updated.

**Comment by dholbach** [comment]
The build fails still: https://travis-ci.org/ubuntu/snappy-playpen/builds/138389626

File: README.md [hunk 4]
```diff
@@ -20,6 +20,7 @@ This project currently includes the following snaps:
 | :white_check_mark:  | `galculator`       |                           |
 | :red_circle:        | `heroku`           |                           |
 | :red_circle:        | `imagemagick6-stable`|                         |
+| :white_check_mark:  | `iota    `         |                           |
```
Comment (dholbach) [inline README.md]: Missing ` and it'd be good to add a few words about which plugins it uses, etc.

**Comment by tachyons** [comment]
Seems like build failure is from atom package , node version is to be updated

**Pull request closed without merging**